rest_framework/routers.py

```diff
@@ -144,7 +144,7 @@ def __init__(self, trailing_slash=True, use_regex_path=True):
             self._url_conf = re_path
         else:
             self._base_pattern = '<{lookup_value}:{lookup_prefix}{lookup_url_kwarg}>'
-            self._default_value_pattern = 'path'
+            self._default_value_pattern = 'str'
```

Comment:
any tests to verify this?

Reply:
Here they are: I have added the corner case in which there may be some patterns which may be ambigouus.

Thus using the `<path:var>` could cause problems, since it is too greedy, and having the latest registered pattern to never match if there is any sub-match.